eureka2-ext/eureka2-eureka1-rest-api/src/main/java/com/netflix/eureka2/eureka1/rest/query/Eureka2RegistryViewCache.java

```diff
@@ -92,6 +87,20 @@ public Applications findApplicationsBySecureVip(final String secureVipAddress) {
         return findApplicationsBy(secureVipAddress, Interests.forSecureVips(secureVipAddress), secureVipMap);
     }
 
+    private RegistryFetch getLatestFullFetch() {
+        if (fullFetchWithDeltaView.get() == null) {
+            Eureka2FullFetchWithDeltaView newView = new Eureka2FullFetchWithDeltaView(
+                    registry.forInterest(Interests.forFullRegistry()),
+                    refreshIntervalMs,
+                    scheduler
+            );
+            if (!fullFetchWithDeltaView.compareAndSet(null, newView)) {
+                newView.close();
+            }
+        }
+        return fullFetchWithDeltaView.get().latestCopy().timeout(queryTimeoutMs, TimeUnit.MILLISECONDS).take(1).toBlocking().first();
```

Comment:
Why toBlocking here? Also, if the timeout occurs and there is no element returned, this would result in an NPE?